include/picongpu/particles/atomicPhysics/rateCalculation/BoundFreeFieldTransitionRates.hpp

```diff
@@ -120,5 +152,118 @@ namespace picongpu::particles::atomicPhysics::rateCalculation
+         */
+        template<
+            typename T_EFieldType,
+            typename T_ChargeStateDataBox,
+            typename T_AtomicStateDataBox,
+            typename T_BoundFreeTransitionDataBox>
+        HDINLINE static float_X rateADKFieldIonization(
+            T_EFieldType const eFieldNorm,
+            float_X const ionizationPotentialDepression,
+            uint32_t const transitionCollectionIndex,
+            T_ChargeStateDataBox const chargeStateDataBox,
+            T_AtomicStateDataBox const atomicStateDataBox,
+            T_BoundFreeTransitionDataBox const boundFreeTransitionDataBox)
+        {
+            if(eFieldNorm == 0._X)
+                return 0._X;
+
+            ResultStruct input = getInput(
+                ionizationPotentialDepression,
+                transitionCollectionIndex,
+                chargeStateDataBox,
+                atomicStateDataBox,
+                boundFreeTransitionDataBox);
+            float_X const nEff = input.effectivePrincipalQuantumNumber;
+            float_X const Z = input.screenedCharge;
+
+            // sim.atomicUnit.eField()
+            float_X const eFieldNorm_AU = sim.pic.conv().eField2auEField(eFieldNorm);
+
+            return rateFormula(Z, nEff, eFieldNorm_AU);
+        }
+
+        /** get maximum field ionization ADK rate for electric field strengths inside the given boundaries
+         *
+         * @tparam T_ChargeStateDataBox instantiated type of dataBox
+         * @tparam T_AtomicStateDataBox instantiated type of dataBox
+         * @tparam T_BoundFreeTransitionDataBox instantiated type of dataBox
+         *
+         * @param maxEFieldNorm maximum E-field vector norm, in sim.units.eField()
+         * @param minEFieldNorm minimum E-field vector norm, in sim.units.eField()
+         * @param ionizationPotentialDepression, in eV
+         * @param transitionCollectionIndex index of transition in boundBoundTransitionDataBox
+         * @param atomicStateDataBox access to atomic state property data
+         * @param boundBoundTransitionDataBox access to bound-bound transition data
+         *
+         * @return unit: 1/picongpu::sim.unit.time()
+         */
+        template<
+            typename T_EFieldType,
+            typename T_ChargeStateDataBox,
+            typename T_AtomicStateDataBox,
+            typename T_BoundFreeTransitionDataBox>
+        HDINLINE static float_X maximumRateADKFieldIonization(
+            T_EFieldType const maxEFieldNorm,
+            T_EFieldType const minEFieldNorm,
+            float_X const ionizationPotentialDepression,
+            uint32_t const transitionCollectionIndex,
+            T_ChargeStateDataBox const chargeStateDataBox,
+            T_AtomicStateDataBox const atomicStateDataBox,
+            T_BoundFreeTransitionDataBox const boundFreeTransitionDataBox)
+        {
+            if(maxEFieldNorm == 0._X)
+                return 0._X;
+
+            ResultStruct input = getInput(
```

Comment:
I think you can choose a very short name for this variable because it doesn't bear any meaning. It's just a hull for the meaningfully named variables `nEff` and `Z`. With the other changes above, this line could look like:
```suggestion
            auto const v = RateFormulaVariables{ //...
```
and usage below
```C++
auto const nEffCubed = pmacc::math::cPow(v.nEff, u32(3u));
```

Reply:
done